providers/utils.go

```
@@ -167,6 +167,27 @@ func (i *ProviderInfo) GetOption(key string) (string, bool) {
 	return val, ok
 }
 
+func (i *ProviderInfo) ReadModuleInfo(moduleID common.ModuleID) ProviderModuleInfo {
+	rootModule := ProviderModuleInfo{
+		BaseURL: i.BaseURL,
+		Support: ModuleSupport{},
+	}
+
+	if i.Modules == nil {
+		// Default to provider information to construct root module.
+		return rootModule
+	}
+
+	modules := *i.Modules
+
+	module, ok := modules[string(moduleID)]
+	if !ok {
+		return rootModule
```

probably a good idea to error out here, because we don't want silent failures when incorrect modules are being passed

---

Good point! Also this will be the single point that will control module validation for all connectors, should be very consistent.